pkg/ovsdb/ovsdb.go

```diff
@@ -166,30 +202,21 @@ func getExternalIDs(row map[string]interface{}) (map[string]string, error) {
 		return nil, fmt.Errorf("row does not contain external_ids")
 	}
 
-	rowValSlice, ok := rowVal.([]interface{})
-	if !ok || len(rowValSlice) != 2 || rowValSlice[0] != "map" {
-		return nil, fmt.Errorf("not a OvsMap: %T: %v", rowVal, rowVal)
-	}
-	mapVals, ok := rowValSlice[1].([]interface{})
+	rowValOvsMap, ok := rowVal.(ovsdb.OvsMap)
 	if !ok {
-		return nil, fmt.Errorf("cannot get map values: %v", rowValSlice[1])
+		return nil, fmt.Errorf("not a OvsMap: %T: %v", rowVal, rowVal)
 	}
-	extIDs := make(map[string]string, len(rowValSlice))
-	for _, mapEntry := range mapVals {
-		me, ok := mapEntry.([]interface{})
-		if !ok || len(me) != 2 {
-			return nil, fmt.Errorf("invalid map entry: %v", mapEntry)
-		}
-		ks := fmt.Sprintf("%v", me[0])
-		vs := fmt.Sprintf("%v", me[1])
-		extIDs[ks] = vs
+
+	extIDs := make(map[string]string, len(rowValOvsMap.GoMap))
+	for key, value := range rowValOvsMap.GoMap {
+		extIDs[key.(string)] = value.(string)
```

Comment:
Are all values granted to be strings? In the original code, there was a conversion happening using Sprintf, so even if the right hand value was non-string value, it would still pass. Can't this new code be broken if somebody sets a value in the map to a number? Disclaimer: I'm not sure if it is possible with OVS.

Reply:
`external_ids` seems to be a pair of strings.

https://github.com/openvswitch/ovs/blob/e5b5008acdf08e90874f5b4da09ffe162fc762aa/vswitchd/vswitch.xml#L38

But I have no problem to use `fmt.Sprintf`.
I fixed it.

Reply:
Oh, if it is guaranteed by the database, let's not duplicate it. Sorry about the noise